sequencer/src/header.rs

```diff
@@ -426,6 +426,50 @@ impl QueryableHeader<SeqTypes> for Header {
     }
 }
 
+impl ExplorerHeader<SeqTypes> for Header {
+    type BalanceAmount = FeeAmount;
+    type WalletAddress = FeeAccount;
+    type ProposerId = FeeAccount;
+    type NamespaceId = NamespaceId;
+
+    fn proposer_id(&self) -> Self::ProposerId {
+        self.fee_info.account()
+    }
+
+    fn fee_info_account(&self) -> Self::WalletAddress {
+        self.fee_info.account()
+    }
+
+    fn fee_info_balance(&self) -> Self::BalanceAmount {
+        self.fee_info.amount()
+    }
+
+    fn reward_balance(&self) -> Self::BalanceAmount {
+        FeeAmount::from(0)
```

Comment:
A comment or issue about what looks like a dummy implementation?

Reply:
This is fair.  Though it does bring up a question.

I don't believe we are currently issuing rewards for constructing `block`s at the moment.  Am I mistaken in this understanding?  If I am, where would we source this reward from?

If we don't intended to add a reward, then creating an issue to address adding a reward would be the incorrect approach, and the explorer ui design would be incorrect. If I am not mistaken, let me know, and I shall add a comment to address this.

Reply:
There's no reward currently, I think we should leave it out of the UI until there is one

Reply:
We can leave it out of the UI, but that's display only.  The question is, do we need to remove it from here as well?  This is just a placeholder value, and the client side is expecting it to be here.  So if we remove the requirement from the `Explorer API`, we will also need to remove it from the parsing on the `Explorer UI`.  Which requires changes in the `Explorer API`, and the `Explorer UI`.  Alternatively if we keep this here, we only have to modify the `Explorer UI` to remove the field from being displayed, and it allows for us to add it later without changing as much code.

As an aside, since we're imposing a fee, but we are not issuing rewards, don't we set ourselves up for the builders eventually running out of the ability to actually build blocks?

Reply:
I'll leave it up to you, personally I don't mind having this placeholder in the backend.

> As an aside, since we're imposing a fee, but we are not issuing rewards, don't we set ourselves up for the builders eventually running out of the ability to actually build blocks?

Yes, the builder will need to be periodically refunded (by bridging from the L1). Although initially the fee will be nominal so the burn rate is very slow. Also, we collect all the burned tokens in a special account so we have the option of distributing rewards retroactively with a protocol upgrade.